### PR TITLE
Remove docker dependency from chromadb tests

### PR DESCRIPTION
**Pull request by poxrud**
# Description

Removed the requirement of running _docker-compose_ for the chromadb tests. This is accomplished by switching to _in-memory_ persistence mode from the _client/server_ mode. This results in much faster test runs, and allows for easier/faster automated testing in a CI/CD environment.  

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by logan-markewich** [comment]
@poxrud Cool, thanks!

File: llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/tests/test_chromadb.py [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
-from typing import Dict, List
+from typing import Dict, List, Generator
 
 import pytest
-from llama_index.core.schema import NodeRelationship, RelatedNodeInfo, TextNode
-from llama_index.core.vector_stores.types import VectorStoreQuery
-from llama_index.vector_stores.chroma import ChromaVectorStore
+from llama_index.schema import NodeRelationship, RelatedNodeInfo, TextNode
+from llama_index.vector_stores import ChromaVectorStore
+from llama_index.vector_stores.types import VectorStoreQuery
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/tests/test_chromadb.py:7]: these imports should be updated to work with v0.10.0, I think you likely just need to revert this section

Comment (poxrud) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/tests/test_chromadb.py]: fixed it

File: llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/tests/test_chromadb.py [hunk 2]
```diff
@@ -29,20 +25,19 @@
...
-    assert isinstance(store, ChromaVectorStore)
-
-
-@pytest.mark.skipif(chromadb_not_available, reason="chromadb is not available")
+# To test chromadb http-client functionality do:
+# cd tests
+# docker-compose up
+#
+@pytest.mark.skipif(
+    http_client_chromadb_mode == False,
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/tests/test_chromadb.py:38]: looks like linting doesnt like testing conditions like this, should be `http_client_chromadb_mode is False`

Comment (poxrud) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/tests/test_chromadb.py]: fixed

**Pull request merged**